internal/controller/discoverymgr/mnedc/server/server.go

```diff
@@ -368,3 +371,23 @@ func (s *Server) Close() error {
 
 	return err
 }
+
+//GetVirtualIP returns the server virtualIP
+func (s *Server) GetVirtualIP() string {
+	return s.virtualIP.String()
+}
+
+//generateServerIP generates a virtual IP for the server
+func generateServerIP(privateIP string) {
+	rand.Seed(time.Now().UnixNano())
+	PrivateIPSplit := strings.Split(privateIP,".")
```

Comment:
Comparing strings is not a bad idea, however I'd like to recommend you to compare IPs with 24 bits of subnet mask. 
You can implement it with package `net` functions such as [ParseIP](https://golang.org/pkg/net/#ParseIP), [CIDRMask(24, 32)](https://golang.org/pkg/net/#CIDRMask) and [(*IPNet) String](https://golang.org/pkg/net/#IPNet.String).
What do you think?